src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs

```diff
@@ -422,6 +422,22 @@ public static async Task<bool> AnyAsync<T>(this ImmutableArray<T> array, Func<T,
             return false;
         }
 
+        public static async Task<bool> AnyAsync<T, TArg>(this ImmutableArray<T> array, Func<T, TArg, Task<bool>> predicateAsync, TArg arg)
```

Comment:
Value task?

Reply:
It probably would be if this were in the runtime, but for us it makes no difference. This form matches the methods already in this file.